flink-ml-lib/src/main/java/org/apache/flink/ml/clustering/kmeans/KMeansModelData.java

```diff
@@ -47,8 +47,16 @@
 
     public DenseVector[] centroids;
 
+    public DenseVector weights;
+
+    public KMeansModelData(DenseVector[] centroids, DenseVector weights) {
+        this.centroids = centroids;
+        this.weights = weights;
+    }
+
     public KMeansModelData(DenseVector[] centroids) {
         this.centroids = centroids;
+        this.weights = new DenseVector(centroids.length);
```

Comment:
It seems a bit weird to have weights to be a vector of 0s.

Would it be simpler to remove this constructor and explicitly specify weights in tests? The weights specify easily by e.g. `Vectors.dense(1, 1)`.

Reply:
I agree. Besides, when users use `OnlineKMeansModel.setModelData` directly, they should be allowed to set weight to `null` or `Vectors.dense()` (empty value), as the prediction process don't care about weights.

Reply:
If this model data is only guaranteed to be used only by `OnlineKMeansModel`, I agree we can do that. 

For the code that constructs a KMeansModelData instance, if it is not guaranteed that this instance is only used by `OnlineKMeansModel`, I suggested to still make this model data `self-contained` with valid weights.